src/count.py

```diff
@@ -0,0 +1,105 @@
+import requests
+import time
+import os 
+import sys
+
+# Constants
+
+GITHUB_API = "https://api.github.com"
+TOKEN = os.getenv('GITHUB_TOKEN')
+HEADERS = {'Authorization': f'token {TOKEN}'}
+MAX_RETRIES = 3
+RETRY_BACKOFF = 2
+
+def safe_request(url, headers, max_retries=MAX_RETRIES, timeout=10):
+    for attempt in range(max_retries):
+        try:
+            response = requests.get(url, headers=headers, timeout=timeout)
+            if response.status_code == 200:
+                return response
+            elif response.status_code == 403 and "X-RateLimit-Reset" in response.headers:
+                reset_time = int(response.headers["X-RateLimit-Reset"])
+                sleep_time = max(reset_time - int(time.time()), 1)
+                print(f"Rate limit exceeded. Waiting for {sleep_time} seconds.")
+                time.sleep(sleep_time)
+            else:
+                print(f"Request failed with status code {response.status_code}. Attempt {attempt + 1} of {max_retries}.")
+                time.sleep(2 ** attempt)  
+        except requests.exceptions.RequestException as e:
+            print(f"Request exception: {e}. Attempt {attempt + 1} of {max_retries}.")
+             # Exponential backoff
+            time.sleep(2 ** attempt) 
+             # Failed to fetch data after retries
+    return None 
+
+def read_repos(file_path):
+    with open(file_path, 'r') as file:
+        repos = [line.strip() for line in file.readlines()]
+    return repos
+
+def get_commits_count(repo):
+    start_date = "2023-01-01T00:00:00Z"
+    end_date = "2024-12-31T23:59:59Z"
+    commits_url = f"{GITHUB_API}/repos/{repo}/commits?since={start_date}&until={end_date}&per_page=1"
+    response = safe_request(commits_url, HEADERS)
+    if response:
+        if 'last' in response.links:
+            last_page_url = response.links['last']['url']
+            last_page_number = int(last_page_url.split('=')[-1])
+            return last_page_number
+        elif response.json():  
+            # Check if there's at least one commit in the specified range
+            return 1
+            # No commits in the specified date range
+        return 0  
+    return "Error"
+
+
+def get_bugs_count(repo):
```

Comment:
why return type is `str | int`, I suggest `Optional[int]` or `Maybe[int]` (see docs for [Maybe](https://returns.readthedocs.io/en/latest/pages/maybe.html))